src/main/frontend/db/rules.cljs

```diff
@@ -29,4 +35,18 @@
     ;;            [(identity ?vs) [?v ...]]
     ;;            (not-join [?e ?v]
     ;;                      [?e ?a ?v]))]
+
+    ;; Recursive optimization
```

Comment:
Perhaps mention child and parent or put it closer to the respective rules? Without more explanation, this comment reads more like a personal comment rather than one meant to provide guidance. Alternatively we could put our notes about datalog perf in lambda